### PR TITLE
Clear all newly added globals when dweet resets

### PR DESCRIPTION
**Pull request by joeytwiddle**
An implementation of [Magna's suggestion](https://github.com/lionleaf/dwitter/issues/93#issuecomment-401560322)

I gave it a little test (with the console log uncommented):

joey s google-chrome at 4 11 04 pm on sunday 1 july 2018

It seems to have detected a var `pauseTime` which we forgot to declare in the dwitter code. So I added a declaration for that too.

This code makes use of `Array.prototype.includes` which is [not supported in IE11](https://caniuse.com/#feat=array-includes). I could fix that if we want to support IE11.

### CODE REVIEW

**Comment by lionleaf** [comment]
Great! Should get rid of most broken dweets. 
I'm guessing code editing still is equally responsive?

(I don't have my laptop at the moment, so will do a proper review / merge later)

Fixes #93

**Comment by joeytwiddle** [comment]
Mmm yeah the code was a bit O(n^2)-ish.

I think it's easier for me to add a couple of lines to make it O(n) than to try to measure performance on a slow device. :stuck_out_tongue: Done!

File: dwitter/templates/dweet/dweet.html [hunk 1]
```diff
@@ -181,7 +182,14 @@
...
     var stats = createStats();
     setStatsVisibility({{ newDweet }});
-  
+
+    // Remember the default globals so that later we can remove any that
+    // were created by the dweet
+    var browserGlobals = {};
+    Object.keys(window).forEach(key => {
+      browserGlobals[key] = true;
+    });
```
Comment (joeytwiddle) [inline dwitter/templates/dweet/dweet.html]: Some "functional programming" people recommend this style instead:
```js
var browserGlobals = Object.keys(window).reduce((a, key) => (a[key] = true, a), {});
```
or even:
```js
var browserGlobals = Object.keys(window).reduce((a, key) => ({...a, [key]: true}), {});
```
I could change to one of those, if you prefer. 😛

Comment (lionleaf) [reply dwitter/templates/dweet/dweet.html]: Fun way of doing it; but I'd say leave it as is now. Easier to read for a greater number of people.

**Pull request merged**